### ISSUE TITLE
"spawned process is not `wait()`ed on in all code paths" when `?` used in chain call before `spawn()`

### ISSUE THREAD

**Issue by lyskov** [rust-lang/rust-clippy#14911]
### Summary

Clippy flag `command.spawn()` code when `?` used in daisy-chain calls _before_ the spawn got called and `.expect` used on spawn invocation.

Interestingly, changing `expect` to `?` on `spawn()` clear the the error:
```rust
fn test_1_passed() -> std::io::Result<String> {
    let (mut recv, send) = std::io::pipe()?;
    let mut command = Command::new("ls").stdout(send.try_clone()?).spawn()?;
    command.wait()?;
    Ok("".into())
}
```

### Lint Name

clippy::zombie_processes

### Reproducer

I tried this code:

```rust
fn test_1_flagged() -> std::io::Result<String> {
    let (mut recv, send) = std::io::pipe()?;
    let mut command = Command::new("ls")
    .stdout(send.try_clone()?)
    .spawn().expect("Could not spawn new process...");
    command.wait()?;
    Ok("".into())
}
```

I expected to see this happen: no issue

Instead, this happened: "spawned process is not `wait()`ed on in all code paths" highlighting `send.try_clone()?` as offending part.

### Version

```text
rustc 1.87.0 (17067e9ac 2025-05-09)
binary: rustc
commit-hash: 17067e9ac6d7ecb70e50f92c1944e545188d2359
commit-date: 2025-05-09
host: aarch64-apple-darwin
release: 1.87.0
LLVM version: 20.1.1
```
